spec/puppet-syntax/templates_spec.rb

```diff
@@ -4,7 +4,8 @@
   let(:subject) { PuppetSyntax::Templates.new }
   let(:conditional_warning_regex) do
     if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.6.0')
-      /2: warning: found `= literal' in conditional/
+      # ruby 3.4 uses '= ,ruby 2.6 to 3.3 uses `=
+      /2: warning: found (`|')= literal' in conditional/
```

Comment:
I'd be tempted to simplify it to:
```suggestion
      /2: warning: found [`']= literal' in conditional/
```